source/Windows.Devices.I2c/I2cDevice.cs

```diff
@@ -41,8 +41,13 @@ internal I2cDevice(string i2cBus, I2cConnectionSettings settings)
                 // save device ID
                 _deviceId = deviceId;
 
-                NativeInit();
-
+                //Instantiate I2C bus once for each bus used
+                if( !I2cController.BusIdCollection.Contains(i2cBus))
+                {
+                    I2cController.BusIdCollection.Add(i2cBus);
```

Comment:
Following this, you need to remove it from the BusIdCollection at `Dispose()` (around line 218). But only if there are no other devices still associated with the BusId we are trying to remove, so the check is a bit tricky...